packages/ui/src/components/Button/Button.tsx

```diff
@@ -25,23 +26,35 @@ export function Button({
             ref={forwardRef}
             type={type}
             className={twMerge(
-                'rounded flex items-center justify-center border border-transparent bg-transparent',
-                size === 'small' ? `p-3 label-medium` : `p-4 label-large`,
+                `${
+                    rounded ? 'rounded-full' : 'rounded'
+                } flex items-center justify-center border border-transparent bg-transparent`,
+                size === 'small'
+                    ? `${rounded ? 'py-2 px-3' : 'p-3'} label-medium`
+                    : `${rounded ? 'py-3 px-4' : 'p-4'} label-large`,
                 variation === 'primary' &&
                     `
                 bg-accent border-accent text-accent-button-text
                 hover:bg-accent-dark hover:border-accent-dark 
-                focus:bg-accent-dark focus:border-transparent focus:ring-2 focus:ring-accent-lighter
-                focus-within:bg-accent-dark focus-within:border-transparent focus-within:ring-2 focus-within:ring-accent-lighter
+                focus:bg-accent-dark focus:border-transparent ${
+                    rounded
+                        ? 'focus:ring-4 focus-within:ring-4'
+                        : 'focus:ring-2 focus-within:ring-2'
+                }  focus:ring-accent-lighter
+                focus-within:bg-accent-dark focus-within:border-transparent focus-within:ring-accent-lighter
                 active:bg-accent-dark active:border-accent-dark
-                disabled:bg-accent-lighter disabled:border-transparent
+                disabled:bg-accent-lighter disabled:border-transparent disabled:opacity-50
                 `,
                 variation === 'secondary' &&
                     `
-                    bg-white border-gray-200 text-gray-800
+                    bg-white ${rounded ? 'border-transparent' : 'border-gray-200'} text-gray-800
```

Comment:
Again, it feels weird that colors of the button change based on the `rounded` property. This is another thing to discuss with the design team.